src/study/system-model/component.cpp

```diff
@@ -0,0 +1,121 @@
+/*
+** Copyright 2007-2024, RTE (https://www.rte-france.com)
+** See AUTHORS.txt
+** SPDX-License-Identifier: MPL-2.0
+** This file is part of Antares-Simulator,
+** Adequacy and Performance assessment for interconnected energy networks.
+**
+** Antares_Simulator is free software: you can redistribute it and/or modify
+** it under the terms of the Mozilla Public Licence 2.0 as published by
+** the Mozilla Foundation, either version 2 of the License, or
+** (at your option) any later version.
+**
+** Antares_Simulator is distributed in the hope that it will be useful,
+** but WITHOUT ANY WARRANTY; without even the implied warranty of
+** MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+** Mozilla Public Licence 2.0 for more details.
+**
+** You should have received a copy of the Mozilla Public Licence 2.0
+** along with Antares_Simulator. If not, see <https://opensource.org/license/mpl-2-0/>.
+*/
+
+#include <ranges>
+
+#include <antares/study/system-model/component.h>
+
+namespace Antares::Study::SystemModel
+{
+
+Component::Component(const ComponentData& data)
+{
+    // Check that mandatory attributes are not empty
```

Comment:
Why not move all these checks on incoming data in a private member function ?
```C++
isComponentDataValid(const ComponentData& data);
```
The current constructor's body would be more simple.

Reply:
ok

Reply:
or even better : in a free fonction, as a method would be useless here.

Reply:
can you read my last commit ?

Reply:
**checkComponentDataValidity** should be a free function : 

```C++
static void checkComponentDataValidity(const ComponentData& data)
{
    ...
}
```